internal/metrics/DBStorage.go

```diff
@@ -0,0 +1,362 @@
+package metrics
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"gmetrics/internal/logger"
+	"time"
+)
+
+type SQLExecutor interface {
+	ExecContext(ctx context.Context, query string, args ...any) (sql.Result, error)
+	QueryRowContext(ctx context.Context, query string, args ...any) *sql.Row
+	QueryContext(ctx context.Context, query string, args ...any) (*sql.Rows, error)
+	BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
+}
+
+// DBStorage хранилище метрик в базе данных
+// TODO Возможно сделать хранилище с промежуточным хранение в памяти
+type DBStorage struct {
+	// storeCtx контекст, который отвечает за запросы
+	storeCtx context.Context
+	// db пул соединений с базой данных, которыми может пользоваться хранилище
+	db SQLExecutor
+}
+
+// NewDBStorage создание нового хранилища в базе данных
+func NewDBStorage(ctx context.Context, db *sql.DB) (*DBStorage, error) {
```

Comment:
Думаю, раз мы определили интерфейс мы хотим его принимать и как параметр тоже?
```suggestion
func NewDBStorage(ctx context.Context, db SQLExecutor) (*DBStorage, error) {
```

Reply:
Сделал принимающий параметр интерфейсом